app/services/versioned_files_service/migrate_action.rb

```diff
@@ -9,26 +9,34 @@ def initialize(object:, version_metadata:)
     end
 
     def call
-      # For each shelved file in the cocina object, make sure there is a content file.
-      check_content_files!
-      # For each shelved file, create a hardlink named by md5
-      link_content_files
-      # Write the cocina to cocina path for the version and create a new head cocina symlink.
-      write_cocina(version: 1, cocina: cocina_hash, head_version: true)
-      # Write the public xml to public xml path for the version and create a new head public xml symlink .
-      write_public_xml(version: 1, public_xml:, head_version: true)
-      copy_meta_json
-      # Update the version manifest.
-      version_manifest.update_version(version: 1, version_metadata:, head_version: true)
+      create_directory_structure
```

Comment:
I don't think we should create the versions directory before checking the content files. Once the versions directory is created, applications will think that the object is using the versioned layout, even though the directory is empty.

Reply:
If you just switch `check_content_files!` and `create_directory_structure` I think the logic should be fine.

Reply:
Except the lock file is created within the AWFL directory structure?